lib/auth/auth_with_roles.go

```diff
@@ -3400,17 +3406,45 @@ func (a *ServerWithRoles) CreateUser(ctx context.Context, user types.User) (type
 	if err := a.action(apidefaults.Namespace, types.KindUser, types.VerbCreate); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	if a.hasBuiltinRole(types.RoleOkta) && !hasOriginOkta(user) {
+		return nil, trace.BadParameter("Okta service must supply okta origin")
+	}
```

Comment:
```suggestion
	} else if ! a.hasBuiltinRole(types.RoleOkta) && hasOriginOkta(user) {
		return nil, trace.BadParameter("Must be Okta service to set Okta origin")
	}
```

This check can be optimized into a single condition if we are ok with changing the error response (`a.hasBuiltinRole(types.RoleOkta) != hasOriginOkta(user)`), but I think we should also be verifying against this condition.